kornia/morphology/morphology.py

```diff
@@ -74,9 +80,13 @@ def dilation(
         neighborhood = structuring_element.clone()
```

Comment:
Why do we need to clone? Potentially be not updatable whilst optimization?

Reply:
Otherwise it is not differentiable

Reply:
why ? this is a weird one

Reply:
Because below we are in-lace replacing the values